src/napari_omero/plugins/loaders.py

```diff
@@ -241,3 +241,192 @@ def get_lazy_big_plane(level_id, level_desc, z, c, t):
         pyramid.append(da.stack(t_stacks))
 
     return pyramid
+
+
+def load_rois(conn: BlitzGateway, image: ImageWrapper) -> list[LayerData]:
+    """Retrieve ROIs from an OMERO image and format them as a single Napari layer."""
+    roi_service = conn.getRoiService()
+    result = roi_service.findByImage(image.getId(), None)
+    img_id = image.getId()
+
+    # Lists to store properties for all shapes
+    all_coords = []
+    all_shape_types = []
+    all_comments = []
+    all_roi_ids = []
+    all_shape_ids = []
+    all_edge_colors = []
+    all_face_colors = []
+
+    # Get Z and T dimension ranges for the image
+    size_z = range(image.getSizeZ())
+    size_t = range(image.getSizeT())
+
+    # Get physical pixel sizes; default to 1 if not defined
+    size_x = image.getPixelSizeX() or 1
+    size_y = image.getPixelSizeY() or 1
+    pixel_size_z = image.getPixelSizeZ() or 1
+
+    # Loop through all ROIs
+    for roi in result.rois:
+        roi_id = roi.getId().getValue()
+
+        # Loop through all shapes in each ROI
+        for shape in roi.copyShapes():
+            if shape is None:
+                continue  # Skip empty shapes
+
+            # Parse shape to Napari-compatible format
+            parsed = parse_omero_shape(shape)
+            if parsed is None:
+                continue  # unsupported shape
+
+            coords_2d, meta, _ = parsed
+            coords_2d = [[y, x] for y, x in coords_2d]
+
+            shape_id = shape.getId().getValue()
+            labels = shape.getTextValue()
+            comment = labels.getValue() if labels else ""
+
+            # Use shape's Z/T if set, else apply to all slices (T, Z)
+            theZ = shape.getTheZ()
+            z_values = [theZ.getValue()] if theZ else list(size_z)
+            theT = shape.getTheT()
+            t_values = [theT.getValue()] if theT else list(size_t)
+
+            # Make meshgrid of (t, z)
+            T, Z = np.meshgrid(t_values, z_values, indexing="ij")  # (T, Z)
+            tz = np.stack([T.ravel(), Z.ravel()], axis=1)  # (T*Z, 2)
+            count = len(tz)
+
+            edge_color = shape.getStrokeColor()
+            fill_color = shape.getFillColor()
+            if not load_points:
+                # Parse shape to Napari-compatible format
+                parsed = parse_omero_shape(shape)
+                if parsed is None:
+                    show_warning(f"Unsupported OMERO shape skipped: {sh_type}")
```

Comment:
`sh_type` isn't defined anywhere - maybe that's a cheap way of doing it?
```suggestion
                    show_warning(f"Unsupported OMERO shape skipped: {type(shape)}")
```